src/Handler/TagHandler.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\Handler;
+
+use FOS\HttpCache\CacheInvalidator;
+use FOS\HttpCache\Exception\UnsupportedProxyOperationException;
+
+/**
+ * Handler for cache tagging.
+ *
+ * @author David de Boer <david@driebit.nl>
+ * @author David Buchmann <mail@davidbu.ch>
+ */
+class TagHandler
+{
+    /**
+     * @var CacheInvalidator
+     */
+    private $invalidator;
+
+    /**
+     * @var string
+     */
+    private $tagsHeader;
+
+    /**
+     * Constructor
+     *
+     * @param CacheInvalidator $invalidator The invalidator instance.
+     * @param string           $tagsHeader  Header to use for tags, defaults to X-Cache-Tags.
+     *
+     * @throws UnsupportedProxyOperationException If CacheInvalidator does not support invalidate requests
+     */
+    public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cache-Tags')
+    {
+        if (!$invalidator->supports(CacheInvalidator::INVALIDATE)) {
+            throw UnsupportedProxyOperationException::cacheDoesNotImplement('BAN');
+        }
+        $this->invalidator = $invalidator;
+        $this->tagsHeader = $tagsHeader;
+    }
+
+    /**
+     * Set the HTTP header name that will hold cache tags.
+     *
+     * @param string $tagsHeader
+     *
+     * @return $this
+     *
+     * @deprecated Pass this setting in the header.
+     */
+    public function setTagsHeader($tagsHeader)
+    {
+        $this->tagsHeader = $tagsHeader;
+
+        return $this;
+    }
```

Comment:
Now we have a separate tagging class to which we can pass this value in its constructor, I say we drop this setter and getter. After all, only funny things will happen if users change the tag header value at runtime.